### DISCUSSION TITLE
Streamline your PR workflows with Sentry's Copilot Extension

### DISCUSSION THREAD

**Discussion by gauthamcs** [getsentry/sentry#79515] [Launch Feedback]
We’re excited to announce a major update to the [Sentry Copilot Extension](https://github.com/marketplace/sentry-github-copilot-extension), which now leverages Sentry’s code-level context and Copilot to make your PR workflows better. This update automates key tasks like generating unit tests, detecting bugs, and suggesting fixes—all without leaving your PR workflow.

## Key Features

- **Automated Unit Tests:** Sentry analyzes your code changes and automatically generates a comprehensive set of unit tests. With just one click, you can add these tests to a new branch, review them, and merge them into your feature branch—saving time and improving test coverage.

- **Bug Detection & Root Cause Analysis:** After your PR is merged, Sentry continues to monitor for issues. If a new problem arises, Sentry’s deep code-level context identifies the root cause and surfaces it directly in the PR. This way, you can see exactly where and why the issue occurred, all without leaving GitHub.

- **Autofix Suggestions:** When an issue is detected, Sentry goes beyond just identifying the problem by suggesting fixes tailored to the actual code changes. You can review and apply these fixes through a draft PR, ensuring a smooth and efficient bug resolution process.

----------

![CleanShot 2024-10-21 at 15 08 36](https://github.com/user-attachments/assets/46cddd7f-97ef-4833-80d5-678104fcc489)

----------
## Getting Started 

Here’s what you need to do to get started:
- Ensure your Sentry organization has opted-in to the use of generative AI features (Settings > Legal & Compliance). 
  - Learn more: [AI/ML: Transparency and Choice](https://docs.sentry.io/security-legal-pii/security/ai-ml-policy/#use-of-identifying-data-for-generative-ai-features)
- Enable and configure Suspect Commits for Sentry and your Github repository
- Enable Sentry Autofix from your Sentry Issues feed
- Ensure you have an active Github Copilot subscription (Individual, Business or Enterprise) https://docs.github.com/en/copilot/about-github-copilot/subscription-plans-for-github-copilot

Add the Sentry Copilot Extension from the [GitHub Marketplace](#) to automate your test generation, bug detection, and fixes within your PRs. 

## We want to hear from you!

We’d like to hear from you about this exciting new capability and what else we could be doing here. Please use this discussion thread to share your feedback.

**Comment by michaelmairegger** [comment]
Does it work with on Self-Hosted Sentry too? After authorizing the extension I get redirected to sentry.io 
![{5B00D4C8-C468-4DAA-860E-B37BFC503A06}](https://github.com/user-attachments/assets/634072f4-7e73-4e5a-873a-88c51e5e3acd)

**Reply by gauthamcs** [reply]
Thanks for your interest! At this time, this capability is only available on Sentry SaaS.
